### ISSUE TITLE
Actions are triggered multiple times when multiple conditions are added

### ISSUE THREAD

**Issue by ykalidin** [project-flogo/rules#63]
**Current behavior:**
Actions are called multiple times when there are multiple actions for a rule and a new tuple data is asserted.

**Expected behavior:**
Actions should be fired only once per data.

**Minimal steps to reproduce the problem:**
1)Attached the main.go file and Descriptor file. There are two conditions in main.go which returns true always. Run the main.go file.
Below is the output seen:
```Rule added: [rule1]
Insert data 1
Rule fired: [rule1]
name= Tom
Insert data 2
Rule fired: [rule1]
name= Bob
Rule fired: [rule1]
name= Bob
Rule fired: [rule1]
name= Tom
```

The rule is fired once when data 1 "Tom" is asserted.
But when the next data "Bob" is asserted, the rule is fired 3 times(2 times for Bob and 1 time for Tom).
Actions are triggered multiple times when there are multiple conditions.This behavior is unexpected.

[issue1.zip](https://github.com/project-flogo/rules/files/3309828/issue1.zip)

**Comment by balamg** [comment]
#64 fixes this as well

**Comment by ykalidin** [comment]
Working as expected.Closing the issue.
